### DISCUSSION TITLE
Create private files

### DISCUSSION THREAD

**Discussion by fleetingbytes** [FiloSottile/age#402] [UX reports]
I wish to decrypt a file such that the output file is private right away.
Currently after `age -o decrypted_secret.txt -i identity.key -d secret.age` I see that the `decrypted_secret.txt` has the default file mode bits `-rw-r--r--` a.k.a. `644`. Right now I need an additional step of `chmod 600 decrypted_secret.txt` to make the decrypted file private. This creates a small time window within which other users could read the contents of the decrypted file before I set it to private. A theoretical `age` option `--private` could avoid this by creating a private file right away.

**Comment by keisentraut** [comment]
This sounds like it is not a problem of ```age``` but sounds like you use the wrong [umask](https://en.wikipedia.org/wiki/Umask) setting. I didn't test it, but have you tried ```umask 0077 && age -d ...```?

**Reply by fleetingbytes** [reply]
Aha, thanks for the pointer. I did not know about umask. A `umask 0077` before the age decryption solves this, of course.
